qiskit_nature/second_q/operators/bosonic_op.py

```diff
@@ -0,0 +1,560 @@
+    non-negative integer representing the index of the bosonic mode where the ``+`` (creation) or
+    ``-`` (annihilation) operation is to be performed. The value of :code:`index` is bound by the
+    number of modes (``num_modes``) of the operator (Note: since Python indices are
+    0-based, the maximum value an index can take is given by :code:`num_modes-1`).
+
+    **Initialization**
+
+    A ``BosonicOp`` is initialized with a dictionary, mapping terms to their respective
+    coefficients:
+
+    .. code-block:: python
+
+        from qiskit_nature.second_q.operators import BosonicOp
+
+        op = BosonicOp(
+            {
+                "+_0 -_0": 1.0,
+                "+_1 -_1": -1.0,
+            },
+            num_modes=2,
+        )
+
+    By default, this way of initializing will create a full copy of the dictionary of coefficients.
+    If you have very restricted memory resources available, or would like to avoid the additional
+    copy, the dictionary will be stored by reference if you disable ``copy`` like so:
+
+    .. code-block:: python
+
+        some_big_data = {
+            "+_0 -_0": 1.0,
+            "+_1 -_1": -1.0,
+            # ...
+        }
+
+        op = BosonicOp(
+            some_big_data,
+            num_modes=2,
+            copy=False,
+        )
+
+
+    .. note::
+
+        It is the users' responsibility, that in the above scenario, :code:`some_big_data` is not
+        changed after initialization of the `BosonicOp`, since the operator contents are not
+        guaranteed to remain unaffected by such changes.
+
+    **Algebra**
+
+    This class supports the following basic arithmetic operations: addition, subtraction, scalar
+    multiplication, operator multiplication, and adjoint.
+    For example,
+
+    Addition
+
+    .. code-block:: python
+
+      BosonicOp({"+_1": 1}, num_modes=2) + BosonicOp({"+_0": 1}, num_modes=2)
+
+    Sum
+
+    .. code-block:: python
+
+      sum(BosonicOp({label: 1}, num_modes=3) for label in ["+_0", "-_1", "+_2 -_2"])
+
+    Scalar multiplication
+
+    .. code-block:: python
+
+      0.5 * BosonicOp({"+_1": 1}, num_modes=2)
+
+    Operator multiplication
+
+    .. code-block:: python
+
+      op1 = BosonicOp({"+_0 -_1": 1}, num_modes=2)
+      op2 = BosonicOp({"-_0 +_0 +_1": 1}, num_modes=2)
+      print(op1 @ op2)
+
+    Tensor multiplication
+
+    .. code-block:: python
+
+      op = BosonicOp({"+_0 -_1": 1}, num_modes=2)
+      print(op ^ op)
+
+    Adjoint
+
+    .. code-block:: python
+
+      BosonicOp({"+_0 -_1": 1j}, num_modes=2).adjoint()
+
+    In principle, you can also add `BosonicOp` and integers, but the only valid case is the
+    addition of `0 + BosonicOp`. This makes the `sum` operation from the example above possible
+    and it is useful in the following scenario:
+
+    .. code-block:: python
+
+        boson = 0
+        for i in some_iterable:
+            # some processing
+            boson += BosonicOp(somedata)
```

Comment:
```suggestion
```
Since starting out from the `FermionicOp` to make this class, I updated that class' docstring and
removed this section. See #1112 for more details.

Reply:
Suggestion applied